airbyte-ci/connectors/metadata_service/orchestrator/orchestrator/jobs/registry.py

```diff
@@ -16,3 +16,34 @@
     selection=registry_entry_inclusive,
     partitions_def=registry_entry.metadata_partitions_def,
 )
+
+
+@op(required_resource_keys={"all_metadata_file_blobs"})
+def add_new_metadata_partitions_op(context):
+    """
+    This op is responsible for polling for new metadata files and adding their etag to the dynamic partition.
+    """
+    MAX_RUN_REQUEST = 50
+    all_metadata_file_blobs = context.resources.all_metadata_file_blobs
+    partition_name = registry_entry.metadata_partitions_def.name
+
+    new_etags_found = [
+        blob.etag for blob in all_metadata_file_blobs if not context.instance.has_dynamic_partition(partition_name, blob.etag)
+    ]
+
+    context.log.info(f"New etags found: {new_etags_found}")
+
+    if not new_etags_found:
+        return SkipReason(f"No new metadata files to process in GCS bucket")
+
+    # if there are more than the MAX_RUN_REQUEST, we need to split them into multiple runs
+    if len(new_etags_found) > MAX_RUN_REQUEST:
+        new_etags_found = new_etags_found[:MAX_RUN_REQUEST]
+        context.log.info(f"Only processing first {MAX_RUN_REQUEST} new blobs: {new_etags_found}")
+
+    context.instance.add_dynamic_partitions(partition_name, new_etags_found)
+
+
+@job
+def add_new_metadata_partitions():
+    add_new_metadata_partitions_op()
```

Comment:
does this need to be added into the list of jobs in the init file, or does it work differently than other jobs?

Reply:
It didnt need to! 

You can see it in this loom video: https://www.loom.com/share/22025644eea24b6d901c635b71c0c68b

But Ill go and add it!

Reply:
... weird! i'd be interested to know if our list has any meaning then :D